shared/reducers/pinentry.js

```diff
@@ -77,7 +77,7 @@ export default function (state: RootPinentryState = initialState, action: Pinent
       }
       return state
     default:
-      if (state.started && sessionID != null) {
+      if (state.started && sessionID != null && isPinentryAction(action)) {
```

Comment:
The end result is an empty pinentry window with a broken render because it has no data.